torch/package/package_importer.py

```diff
@@ -337,20 +393,56 @@ def _make_module(
 
         return module
 
+    def _check_if_viable_parent(self, cur, child, name):
+        if isinstance(cur, _SelectiveExternNode):
+            return
+        if not isinstance(cur, (_PackageNode)) or child not in cur.children:
+            raise ModuleNotFoundError(
+                f'No module named "{name}" with in self-contained archive "{self.filename}"'
+                f" and the module is also not in the list of allowed external modules: {self.extern_modules}",
+                name=name,
+            )
+
+    def _get_node_from_name(self, name: str):
+        cur: _PathNode = self.root
+        atoms = name.split(".")
+        for atom in atoms[:-1]:
+            if isinstance(cur, _ExternNode):
+                return cur
+            if isinstance(cur, (_SelectiveExternNode, _PackageNode)):
+                self._check_if_viable_parent(cur, atom, name)
+                cur = cur.children[atom]
+        if isinstance(cur, (_SelectiveExternNode, _PackageNode)):
+            self._check_if_viable_parent(cur, atoms[-1], name)
+            return cur.children[atoms[-1]]
+        elif isinstance(cur, _ExternNode):
```

Comment:
When would this ever happen? `_ExternNode` should be fulfilled by `importlib.import_module`, and should never be reached from `_make_module`.

Reply:
Yep in the current iteration in never happens, I was intending `_get_node_from_name` to be a general purpose helper function. So I think it's worth keeping in. I will add an assert in `_make_module` to make sure things are working as intended.